python/python-psi-impl/src/com/jetbrains/python/refactoring/classes/PyClassRefactoringUtil.java

```diff
@@ -518,8 +521,31 @@ public static void optimizeImports(@NotNull final PsiFile file) {
     PyImportOptimizer.onlyRemoveUnused().processFile(file).run();
   }
 
+
+  public static PsiFile placeFile(Project project, String path, String filename) throws IOException {
```

Comment:
This overload of `placeFile` is still used in `PyExtractSuperclassHelper` and in the "Extract superclass" refactoring, which still causes the same problem with unnecessary `__init__.py` created inside namespace packages. Please take namespace packages into account there as well, add regression tests and remove this overload entirely.